service/worker/scheduler/workflow.go

```diff
@@ -1144,6 +1146,12 @@ func (s *scheduler) updateMemoAndSearchAttributes() {
 	}
 }
 
+func (s *scheduler) deterministicProtoEqual(a, b *schedulepb.ScheduleListInfo) bool {
+	ab, _ := proto.Marshal(a)
+	bb, _ := proto.Marshal(b)
+	return string(ab) == string(bb)
```

Comment:
This is wrong:
proto.Marshal is not guaranteed to be deterministic unless you use MarshalOptions{Deterministic:true}. But if you're doing that there's no reason not to use proto.Equal, see https://github.com/protocolbuffers/protobuf-go/blob/master/proto/equal.go#L40-L41


(Also, don't convert to string to compare (which does a copy in most cases), use bytes.Equal)

Reply:
Maybe the real fix here is to update the workflow static checker so that it accepts `proto.Equal`

Reply:
workflowcheck is not infallible, it already has a big ignore list and it's almost certainly missing more stuff that's actually deterministic. It should definitely be updated, but in the meantime let's leave this code correct.